esp-hal/src/lcd_cam/cam.rs

```diff
@@ -482,51 +484,42 @@ impl RxEightBits {
-        pin_5: impl Peripheral<P = P5> + 'd,
-        pin_6: impl Peripheral<P = P6> + 'd,
-        pin_7: impl Peripheral<P = P7> + 'd,
-    ) -> Self
-    where
-        P0: PeripheralInput,
-        P1: PeripheralInput,
-        P2: PeripheralInput,
-        P3: PeripheralInput,
-        P4: PeripheralInput,
-        P5: PeripheralInput,
-        P6: PeripheralInput,
-        P7: PeripheralInput,
-    {
-        crate::into_ref!(pin_0);
-        crate::into_ref!(pin_1);
-        crate::into_ref!(pin_2);
-        crate::into_ref!(pin_3);
-        crate::into_ref!(pin_4);
-        crate::into_ref!(pin_5);
-        crate::into_ref!(pin_6);
-        crate::into_ref!(pin_7);
-
-        pin_0.init_input(Pull::None, crate::private::Internal);
-        pin_0.connect_input_to_peripheral(InputSignal::CAM_DATA_0, crate::private::Internal);
-        pin_1.init_input(Pull::None, crate::private::Internal);
-        pin_1.connect_input_to_peripheral(InputSignal::CAM_DATA_1, crate::private::Internal);
-        pin_2.init_input(Pull::None, crate::private::Internal);
-        pin_2.connect_input_to_peripheral(InputSignal::CAM_DATA_2, crate::private::Internal);
-        pin_3.init_input(Pull::None, crate::private::Internal);
-        pin_3.connect_input_to_peripheral(InputSignal::CAM_DATA_3, crate::private::Internal);
-        pin_4.init_input(Pull::None, crate::private::Internal);
-        pin_4.connect_input_to_peripheral(InputSignal::CAM_DATA_4, crate::private::Internal);
-        pin_5.init_input(Pull::None, crate::private::Internal);
-        pin_5.connect_input_to_peripheral(InputSignal::CAM_DATA_5, crate::private::Internal);
-        pin_6.init_input(Pull::None, crate::private::Internal);
-        pin_6.connect_input_to_peripheral(InputSignal::CAM_DATA_6, crate::private::Internal);
-        pin_7.init_input(Pull::None, crate::private::Internal);
-        pin_7.connect_input_to_peripheral(InputSignal::CAM_DATA_7, crate::private::Internal);
+    pub fn new<'d>(
+        pin_0: impl Peripheral<P = impl PeripheralInput> + 'd,
+        pin_1: impl Peripheral<P = impl PeripheralInput> + 'd,
+        pin_2: impl Peripheral<P = impl PeripheralInput> + 'd,
+        pin_3: impl Peripheral<P = impl PeripheralInput> + 'd,
+        pin_4: impl Peripheral<P = impl PeripheralInput> + 'd,
+        pin_5: impl Peripheral<P = impl PeripheralInput> + 'd,
+        pin_6: impl Peripheral<P = impl PeripheralInput> + 'd,
+        pin_7: impl Peripheral<P = impl PeripheralInput> + 'd,
+    ) -> Self {
+        crate::into_mapped_ref!(pin_0);
+        crate::into_mapped_ref!(pin_1);
+        crate::into_mapped_ref!(pin_2);
+        crate::into_mapped_ref!(pin_3);
+        crate::into_mapped_ref!(pin_4);
+        crate::into_mapped_ref!(pin_5);
+        crate::into_mapped_ref!(pin_6);
+        crate::into_mapped_ref!(pin_7);
+
+        const SIGNALS: [InputSignal; 8] = [
+            InputSignal::CAM_DATA_0,
+            InputSignal::CAM_DATA_1,
```

Comment:
This makes easier to make sure no pins are missed or mismatched.
```suggestion
        let SIGNALS: [_; 8] = [
            (InputSignal::CAM_DATA_0, pin_0),
            (InputSignal::CAM_DATA_1, pin_1),
```